server/src/controllers/Users/resolver.ts

```diff
@@ -1,47 +1,102 @@
 import { Arg, Authorized, Int, Mutation, Query, Resolver } from 'type-graphql';
 
+import { Prisma } from '@prisma/client';
 import { prisma } from '../../prisma';
 
 import { UserWithInstanceRole } from '../../graphql-types';
 import { Permission } from '../../../../common/permissions';
+import { ChapterRoles } from '../../../prisma/generator/factories/chapterRoles.factory';
+import { InstanceRoles } from '../../../prisma/generator/factories/instanceRoles.factory';
+
+const instanceRoleInclude = {
+  instance_role: {
+    include: {
+      instance_role_permissions: {
+        include: { instance_permission: true },
+      },
+    },
+  },
+};
+
+type UserWithUserChapters = Prisma.usersGetPayload<{
+  include: {
+    user_chapters: { include: { chapter_role: true } };
+  };
+}>;
+
+interface ChangeRoleNameData {
+  oldRole: string;
+  newRole: string;
+  user: UserWithUserChapters;
+}
+
+const getRoleName = ({ oldRole, newRole, user }: ChangeRoleNameData) => {
+  if (
+    oldRole === InstanceRoles.chapter_administrator &&
+    newRole === InstanceRoles.member
+  )
+    return InstanceRoles.chapter_administrator;
```

Comment:
```suggestion
  // The client should not request this change, but if it does it's ignored.
  if (
    oldRole === InstanceRoles.chapter_administrator &&
    newRole === InstanceRoles.member
  )
    return InstanceRoles.chapter_administrator;
```
Some context would help